k8s/templates/gc-cron-job.yaml

```diff
@@ -0,0 +1,59 @@
+---
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ template "studio.fullname" . }}-contentnode-gc
+spec:
+  schedule: "0 23 * * 0"
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            app: {{ template "studio.fullname" . }}
+            environment: {{ .Release.Name }}
+        spec:
+          # if gc somehow takes 1 week, replace the running gc with a new one
+          concurrencyPolicy: Replace 
+          # turn off any new cron jobs from running by setting suspend to true.
+          # this is how we toggle this job on and off.
+          suspend: false
+          restartPolicy: Never
+          containers:
+          - name: runner
+            image: {{ .Values.studioApp.imageName }}
+            command: ["make", "contentnodegc"]
+            env:
```

Comment:
All the other things look good to me! I'm just wondering why we are not using the vars from `studio.sharedEnvs` here. It seems that the variables we define here are included in `studio.sharedEnvs`?

Reply:
Ah yes, I struggled with this one :) The whole reason I pulled out the shared envs is so I can actually use it here.

But helm's templating is weird in that the template expansion of studio.sharedEnvs would also copy the indentation found in helpers.tpl. That worked for the deployment and the worker manifest, but the cron job's env location is slightly more nested, making the syntax invalid if I put in studio.sharedEnvs in there.

So I just went with just copying over only the envs that I needed. Which is also safer on our side, since the cron job only has access to the environment variables it needs.

Reply:
ah that makes sense to me! thank you for the explanation 👍